src/main/java/usi/si/seart/staticcodeanalysis/TerminalExecution.java

```diff
@@ -0,0 +1,157 @@
+import java.util.Map;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+/**
+ * A useful wrapper for managing the lifecycle of terminal processes.
+ */
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class TerminalExecution {
+
+    Process process;
+
+    ProcessBuilder processBuilder;
+
+
+    /**
+     * Prepares a new terminal command.
+     * @param cwd The Current Working Directory in which the command will be invoked.
+     * @param args the arguments of the command. Will get joined together (eg {"ls", "-la"} )
+     */
+    public TerminalExecution(@Nullable Path cwd, @NotNull String... args) {
+        processBuilder = new ProcessBuilder();
+
+        if (cwd != null)
+            processBuilder.directory(cwd.toFile());
+
+        switch (System.getProperty("os.name")) {
+            case "windows":
+                processBuilder.command("cmd.exe", "/c", String.join(" ", args));
+                break;
+            default:
+                processBuilder.command("bash", "-c", String.join(" ", args));
+                break;
+        }
+    }
+
+
+    /**
+     * Invokes the command in the shell.
+     *  If a process was already invoked and is still running, it first stops it.
+     * @throws TerminalExecutionException Thrown if an error occurred when invoking the command.
+     */
+    public TerminalExecution start() throws TerminalExecutionException {
+        if (isRunning())
+            stop();
+        try {
+            process = processBuilder.start();
+        } catch (IOException e) {
+            throw new TerminalExecutionException("Could not start terminal process",e);
+        }
+        return this;
+    }
+
+    /**
+     * Terminates the process.
+     *  If the process is not running, does nothing.
+     */
+    public TerminalExecution stop() {
+        if(process == null)
+            return this;
+        process.destroy();
+        return this;
+    }
+
+    /**
+     * Returns a BufferedReader for reading from the process' normal output.
+     * @return the BufferedReader for reading the normal output piped from the process.
+     */
+    public BufferedReader getStdout() {
+        return new BufferedReader(new InputStreamReader(process.getInputStream()));
+    }
+
+
+    public BufferedReader getStdErr() {
+        return new BufferedReader(new InputStreamReader(process.getErrorStream()));
+    }
+
+    public boolean isRunning() {
+        return process != null && process.isAlive();
+    }
+
+    /**
+     * Waits for the process to end.
+     *  Method returns without error regardless of the process terminating abnormally, or if the waiting thread is interrupted.
+     */
+    public TerminalExecution waitEnd() {
+        if (process == null)
+            return this;
+
+        try {
+            process.onExit().get();
+        } catch (ExecutionException | CancellationException e) {
+
```

Comment:
Empty `catch`? You might as well `SneakyThrows` at that point, or log the exception if the message will help the programmer